FrontPage/AppDelegate.swift

```diff
@@ -38,6 +40,32 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
   
   func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
     apollo.cacheKeyForObject = { $0["id"] }
+    
+    // bootstrap the registration process by asking the user to 'Accept' and then register with APNS thereafter
+    let settings = UIUserNotificationSettings(types: [.alert, .badge, .sound], categories: nil)
+    UIApplication.shared.registerUserNotificationSettings(settings)
+    UIApplication.shared.registerForRemoteNotifications()
+    
+    // Display all push messages (even the message used to open the app)
+    if let options = launchOptions {
+      if let option = options[UIApplication.LaunchOptionsKey.remoteNotification] as? [String: Any] {
+        let defaults: UserDefaults = UserDefaults.standard;
+        // Send a message received signal to display the notification in the table.
+        if let aps = option["aps"] as? [String: Any] {
+          if let alert = aps["alert"] as? String {
+            defaults.set(alert, forKey: "message_received")
+            defaults.synchronize()
+          } else {
+            if let alert = aps["alert"] as? [String: Any] {
+              let msg = alert["body"]
+              defaults.set(msg, forKey: "message_received")
+              defaults.synchronize()
```

Comment:
why do we save these messages? is the saved data used any where?

Reply:
No we do not, I will remove